### PR TITLE
feat: in-app ramps for private embed feature

### PR DESCRIPTION
**Pull request by ankitsrivas14**
Added in-app ramp in the embed modal and app settings for private embed feature of Business Edition.

#### PR fixes following issue(s)
Fixes # https://github.com/appsmithorg/cloud-services/issues/961

#### Media
Self QA - CE (DP) : https://www.loom.com/share/2eef75fcaee54f7e9fe2441d1be1da00
Self QA - EE (Local) : https://www.loom.com/share/74d23071177244aaae685b31a999a0c6
>
>
#### Type of change

- New feature (non-breaking change which adds functionality)

>
>
>
- [x] Manual
- [x] Jest
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/cypress/support/Pages/HomePage.ts [hunk 1]
```diff
@@ -254,6 +254,7 @@ export class HomePage {
 
   //Maps to AppSetupForRename in command.js
   public RenameApplication(appName: string) {
+    this.onboarding.closeIntroModal();
```
Comment (dipyamanbiswas07) [inline app/client/cypress/support/Pages/HomePage.ts:257]: is this change required?

File: app/client/src/pages/Applications/EmbedSnippet/PrivateEmbeddingContent.tsx [hunk 2]
```diff
@@ -44,31 +37,72 @@ function PrivateEmbeddingContent(props: {
...
-        rel="noreferrer"
-        to="#"
-      >
-        {createMessage(IN_APP_EMBED_SETTING.appsmithBusinessEdition)}
-      </StyledLink>
-    </Text>
-  );
+export function PrivateEmbedRampModal() {
+  const rampLinkSelector = getRampLink({
+    section: "share_modal",
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Applications/EmbedSnippet/PrivateEmbeddingContent.tsx:42]: can we add enums for this instead of  a plain string?

File: app/client/src/pages/Applications/EmbedSnippet/PrivateEmbeddingContent.tsx [hunk 3]
```diff
@@ -44,31 +37,72 @@ function PrivateEmbeddingContent(props: {
...
+        </Link>
+      </div>
+    );
+  }
+  return null;
+}
+export function PrivateEmbedRampSidebar() {
+  const rampLinkSelector = getRampLink({
+    section: "app_settings",
+    feature: "private_embeds",
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Applications/EmbedSnippet/PrivateEmbeddingContent.tsx:83]: same as above

File: app/client/cypress/support/Pages/HomePage.ts (hunk 1 above)
Comment (ankitsrivas14) [reply app/client/cypress/support/Pages/HomePage.ts:257]: Yeah, this was added by Aishwarya because test case was failing as intro modal was not closed.

File: app/client/src/pages/Applications/EmbedSnippet/PrivateEmbeddingContent.tsx (hunk 2 above)
Comment (ankitsrivas14) [reply app/client/src/pages/Applications/EmbedSnippet/PrivateEmbeddingContent.tsx]: Done

File: app/client/src/pages/Applications/EmbedSnippet/PrivateEmbeddingContent.tsx (hunk 3 above)
Comment (ankitsrivas14) [reply app/client/src/pages/Applications/EmbedSnippet/PrivateEmbeddingContent.tsx]: Done

**Review by dipyamanbiswas07 — DISMISSED** [review]
LGTM

**Review by dipyamanbiswas07 — APPROVED** [review]
LGTM
